extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -334,6 +334,10 @@ void InvokeHTTP::onTriggerWithClient(const std::shared_ptr<core::ProcessContext>
 
   logger_->log_debug("onTrigger InvokeHTTP with %s to %s", client.getRequestMethod(), client.getURL());
 
+  const auto remove_callback_from_client_at_exit = gsl::finally([&client] {
+    client.setUploadCallback({});
```

Comment:
I feel like this is the responsibility of `ResourceQueue` and should be handled in `ResourceQueue::returnResource`, what I am thinking is a `resetter: std::function<void(ResourceType&)>` (probably an extra constructor argument to `ResourceQueue`) which when set would be called in `returnResource`, what do you think?

Reply:
we can address this and any related issues with ResourceQueue in https://issues.apache.org/jira/browse/MINIFICPP-1973